tests/hypothesis_strategies/objectives.py

```diff
@@ -0,0 +1,33 @@
+"""Hypothesis strategies for objectives."""
+
+import hypothesis.strategies as st
+
+from baybe.objectives.desirability import DesirabilityObjective
+from baybe.objectives.enum import Scalarizer
+from baybe.objectives.single import SingleTargetObjective
+
+from ..hypothesis_strategies.targets import numerical_targets
+from ..hypothesis_strategies.utils import intervals as st_intervals
+
+
+def single_target_objectives():
+    """Generate :class:`baybe.objectives.single.SingleTargetObjective`."""
+    return st.builds(SingleTargetObjective, target=numerical_targets())
+
+
+@st.composite
+def desirability_objectives(draw: st.DrawFn):
+    """Generate :class:`baybe.objectives.desirability.DesirabilityObjective`."""
+    intervals = st_intervals(exclude_fully_unbounded=True, exclude_half_bounded=True)
+    targets = draw(
+        st.lists(numerical_targets(intervals), min_size=2, unique_by=lambda t: t.name)
+    )
+    weights = draw(
+        st.lists(
+            st.floats(min_value=0.0, exclude_min=True),
```

Comment:
do we explicitly want to allow a (nonsensical) weight of 0.0 ? conceptually this target then should simply be ignored, but maybe there could be numerical problems, e.g. in the geom_mean or mean utility functions?

Reply:
Have a slight tendency to not allowing weights of 0 for the reasons you mention, and also because I do not see a real use case or reason for having weights of 0.

Reply:
However, I think this is currently not being considered in the creation of objectives, right? If I see it correctly, users are able to provide a weight of 0 which is at the very least not consistent with the test here.

Reply:
Could it be that you simply overlooked the `exclude_min=True`?
BTW: stuff is now validated in the objectives, i.e. it's impossible to provide a zero weight:
<img width="541" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/dc5cf1df-9566-44d3-ae02-3b6fa273bd3c">

Reply:
Hi @AVHopp, I think we responded at the exact same moment so I didn't see your message until now. Why do you think users are able to provide 0.0 as weights?

Reply:
>> Could it be that you simply overlooked the exclude_min=True?

looked at it twice and still overlooked it :D
Resolved